### ISSUE TITLE
`DropdownMenu` in `Dialog` not actually in the dialog

### ISSUE THREAD

**Issue by Thijsiez** [JetBrains/compose-jb#1372]
So for this small project I'm working on, I'm running into an issue with using `DropdownMenu`s in `Dialog`s. The problem is that the dropdown menu shows up "under" the dialog and in the main window, not actually where's it's supposed to be; in the dialog. I wrote the most minimal example I could to demonstrate the effect. Thanks for all the work on this awesome library!!

Tested in `beta3` and `beta6-dev446`
```
import androidx.compose.foundation.layout.Box
import androidx.compose.material.Button
import androidx.compose.material.DropdownMenu
import androidx.compose.material.DropdownMenuItem
import androidx.compose.material.Text
import androidx.compose.runtime.*
import androidx.compose.ui.window.Dialog
import androidx.compose.ui.window.Window
import androidx.compose.ui.window.application

fun main() = application {
    var showDialog by remember { mutableStateOf(false) }

    Window(
        onCloseRequest = ::exitApplication,
        title = "Drowdown in Dialog Test"
    ) {
        Button(
            onClick = { showDialog = true }
        ) {
            Text("Open the dialog")
        }

        if (showDialog) theDialog { showDialog = false }
    }
}

@Composable
fun theDialog(onHide: () -> Unit) {
    var expanded by remember { mutableStateOf(false) }

    Dialog(
        onCloseRequest = onHide
    ) {
        Box {
            Button(
                onClick = { expanded = true }
            ) {
                Text("Open the dropdown")
            }
            DropdownMenu(
                expanded = expanded,
                onDismissRequest = { expanded = false }
            ) {
                DropdownMenuItem(onClick = {}) { Text("First item") }
                DropdownMenuItem(onClick = {}) { Text("Second item") }
                DropdownMenuItem(onClick = {}) { Text("Third item") }
            }
        }
    }
}
```

**Comment by igordmn** [comment]
Thanks!

The issue is reported [here](https://github.com/JetBrains/compose-jb/issues/1329) too. We plan to fix that in 1.0 and it has a critical priority.

**Comment by Thijsiez** [comment]
Can't believe I missed that, sorry! Thanks Igor!
